pkg/capiutil/azuremachinepool.go

```diff
@@ -40,7 +43,14 @@ func FindAzureMachinePoolsForCluster(ctx context.Context, client ctrl.Client, cl
 			return nil, microerror.Mask(err)
 		}
 
-		azureMachinePools = azureMachinePoolList.Items
+		for _, azureMachinePool := range azureMachinePoolList.Items {
+			_, isE2E := azureMachinePool.Labels[E2ENodepool]
```

Comment:
Uh, wait! No! Don't do it! :stop_sign: 

`pkg/capiutil` now has a helper function very specific to single test :grin: 

Maybe add `map[string]string` parameter to accept label selector or alternatively `ListOptions` ?

Reply:
Actually the purpose was the contrary: to make it a general function for all tests, so all tests would ignore node pools labeled with `e2e`. I don't want to select those node pools, I want to ignore them. In that case, if a test needs to create stuff, it can label it with the `e2e` label and it won't affect other tests. Does that make sense?

Reply:
Ok, I guess it makes sense, but would it make sense to denote this in function name somehow? I could actually try to use this function for listing machinepools and then struggle wondering where's my test node pool :see_no_evil:

Reply:
I pushed `FindNonTestingAzureMachinePoolsForCluster`. What do you think?

Reply:
Works for me :+1: